include/flyft/data_layout.h

```diff
@@ -3,24 +3,39 @@
 
 #include <iterator>
 #include <type_traits>
+#include <vector>
 
 namespace flyft
     {
+//! DataLayout
+/*!
+ * Generates data structure for the custom data types used in the software.
+ * The class provides get and set functions for the array index and implements N-dimensional array
+ * mapping into one dimensional array using strided indexing scheme.
+ */
 class DataLayout
     {
     public:
     DataLayout();
-    explicit DataLayout(int shape_);
+    //! Constructor for the DataLayout
+    /*!
+     * \param shape_ Shape of the index array
+     */
+    explicit DataLayout(const std::vector<int>& shape);
 
-    int operator()(int idx) const;
-    int shape() const;
+    // Getter for the class
+    int operator()(const std::vector<int>& idx) const;
+
+    // Setter for shape
+    std::vector<int> shape() const;
```

Comment:
This is not doxygen and it is also not a setter. It's OK, though, to only use a short description for getters.
```suggestion
    //! Shape of the layout in each dimension
    std::vector<int> shape() const;
```